lib/galaxy/model/__init__.py

```diff
@@ -846,7 +846,7 @@ def get_param_values( self, app ):
         dict of tool parameter values.
         """
         param_dict = dict( [ ( p.name, p.value ) for p in self.parent_job.parameters ] )
-        tool = app.toolbox.get_tool( self.tool_id )
+        tool = app.toolbox.get_tool( self.tool_id, tool_version=self.job.tool_version )
```

Comment:
I think you're right. The `Task` thing is for `parallelsim="true"` tools, right ? I don't think we have many of these, so I suppose it's possible we never noticed this.